app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -48,23 +48,38 @@
   private static final long WAIT_INTERVAL = 60 * 1000;      // Interval between fetches and retries
   private static final int EXECUTOR_NUM = 3;                // The number of executor threads to analyse the jobs
 
+  public static final String EXECUTOR_NUM_KEY = "elephant.analysis.thread.num";
+
   private AtomicBoolean _running = new AtomicBoolean(true);
   private long lastRun;
+  private int _executorNum;
   private HadoopSecurity _hadoopSecurity;
   private ExecutorService _service;
   private BlockingQueue<AnalyticJob> _jobQueue;
   private AnalyticJobGenerator _analyticJobGenerator;
+  private Configuration _configuration;
+
+  private void loadGeneralConfiguration() {
+    _configuration = new Configuration();
+    _configuration.addResource(this.getClass().getClassLoader().getResourceAsStream("GeneralConf.xml"));
+
+    try {
+      _executorNum = _configuration.getInt(EXECUTOR_NUM_KEY, EXECUTOR_NUM);
+    } catch (NumberFormatException e) {
+      logger.error("invalid config " + EXECUTOR_NUM_KEY + " value: " + _configuration.get(EXECUTOR_NUM_KEY)
```

Comment:
logger.error("Invalid configuration " + EXECUTOR_NUM_KEY + " in GeneralConf.xml. Value is "
          + _configuration.get(EXECUTOR_NUM_KEY) + ". Resetting it to default value: "
          + EXECUTOR_NUM_NUM);